x-pack/plugins/observability/public/pages/assets_inventory/assets_inventory.tsx

```diff
@@ -0,0 +1,121 @@
+import { usePluginContext } from '../../hooks/use_plugin_context';
+import { ObservabilityPublicPluginsStart } from '../../plugin';
+
+const TestBox: React.FC = ({ children }) => (
+  <div style={{ padding: 20, border: '1px solid magenta' }}>{children}</div>
+);
+
+export function AssetsInventoryPage() {
+  const { ObservabilityPageTemplate } = usePluginContext();
+  const { services } = useKibana<ObservabilityPublicPluginsStart>();
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [hosts, setHosts] = useState<Asset[]>([]);
+
+  useEffect(() => {
+    async function retrieve() {
+      setIsLoading(true);
+      const { hosts } = await services.assetManager.publicAssetsClient.getHosts({ from: 'now-1d' });
+      setHosts(hosts);
+      setIsLoading(false);
+    }
+    retrieve();
+  }, [services]);
+
+  return (
+    <ObservabilityPageTemplate
+      data-test-subj="assetsInventoryPageWithData"
+      pageHeader={{
+        pageTitle: (
+          <>
+            {i18n.translate('xpack.observability.assetsInventoryTitle', {
+              defaultMessage: 'Hosts Inventory',
+            })}{' '}
+          </>
+        ),
+      }}
+    >
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <TestBox>Unified search box</TestBox>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiSpacer />
+      <EuiFlexGroup direction="row" gutterSize="m">
+        <EuiFlexItem grow={false}>
+          <TestBox>
+            <div style={{ width: '100px' }}>Sidebar area</div>
+          </TestBox>
+        </EuiFlexItem>
+        <EuiFlexItem style={{ height: '75vh' }}>
+          {isLoading ? <EuiProgress /> : <AssetsTable assets={hosts} />}
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </ObservabilityPageTemplate>
+  );
+}
+
+function isAssetKey(asset: Asset | undefined, key: string): key is keyof Asset {
+  return typeof asset !== 'undefined' && key in asset;
+}
+
+type ColumnDef = { id: keyof Asset } & EuiDataGridColumn;
+
+function AssetsTable({ assets }: { assets: Asset[] }) {
+  const columns: ColumnDef[] = [
+    {
+      id: '@timestamp',
+      displayAsText: 'Last Seen',
+      schema: 'datetime',
+    },
+    {
+      id: 'asset.ean',
+      displayAsText: 'EAN',
+    },
+    {
+      id: 'asset.kind',
+      displayAsText: 'Kind',
+    },
+    {
+      id: 'cloud.provider',
+      displayAsText: 'Cloud Provider',
+    },
+    {
+      id: 'cloud.region',
+      displayAsText: 'CSP Region',
+    },
+  ];
+
+  return (
+    <EuiDataGrid
+      aria-label="Assets table grid"
+      columns={columns}
+      columnVisibility={{
+        visibleColumns: ['@timestamp', 'asset.ean', 'asset.kind', 'cloud.provider', 'cloud.region'],
+        setVisibleColumns: () => {},
+      }}
+      rowCount={Math.min(assets.length, 100)}
+      renderCellValue={({ rowIndex, columnId }) => {
+        return isAssetKey(assets[rowIndex], columnId) ? `${assets[rowIndex][columnId]}` : null;
```

Comment:
Typescript made me do it! Is that a valid excuse? 😆

Reply:
Maybe :D Did it just require you to add the type guard or also to use `renderCellValue`? Because I don't think you need to use `renderCellValue` here?

Reply:
No, it seems like TS thinks renderCellValue is a required param...

<img width="1138" alt="Screenshot 2023-10-17 at 1 35 36 PM" src="https://github.com/elastic/kibana/assets/159370/d6febb73-0224-41c9-822f-59ec28262c11">

Reply:
Hm I notice that you're not passing in the rows as a prop (`items`), maybe that's the issue leading to the type being used for checking? Or maybe EuiDataGrid works differently 🤷🏼

Reply:
Ah I guess DataGrid works differently from the BasicTable.